tests/testthat/setup-testing_depth.R

```diff
@@ -0,0 +1,49 @@
+#' Returns testing depth set by session option or by environmental variable.
+#'
+#' @details Looks for the session option `TESTING_DEPTH` first. If not set,
+#' takes the system environmental variable `TESTING_DEPTH`. If neither
+#' is set, then returns 3 by default. If the value of `TESTING_DEPTH`
+#' is not a numeric of length 1, then returns 3.
+#'
+#' @return `numeric(1)` the testing depth.
+#'
+get_testing_depth <- function() {
+  default_depth <- 3
+  depth <- getOption("TESTING_DEPTH", Sys.getenv("TESTING_DEPTH", default_depth))
+  depth <- tryCatch(
+    as.numeric(depth),
+    error = function(error) default_depth,
+    warning = function(warning) default_depth
+  )
+  if (length(depth) != 1) depth <- default_depth
```

Comment:
@chlebowa I think styler
https://github.com/insightsengineering/teal/blob/main/.github/workflows/check.yaml#L86C11-L86C45
runs simple `styler::style_file` without any `transformations` https://github.com/insightsengineering/r.pkg.template/blob/main/.github/workflows/style.yaml#L129
hence this will not be changed
<img width="596" alt="image" src="https://github.com/insightsengineering/teal/assets/133694481/8538e6f5-2d0f-4686-b69d-ff7476720793">